### ISSUE TITLE
[Feature Request] Support for installing mods from Github releases

### ISSUE THREAD

**Issue by RealCamar** [PolyMC/PolyMC#424]
### Role

Playing modded Minecraft and managing mods 

### Suggestion

Add support for installing/updating mods from Github releases 

### Benefit

So that I can download mods from Github releases (that don’t use CF/Modrinth) without it being time-consuming

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

Maybe a good way of starting to implement this is manually specifying the Github URL

**Comment by forkiesassds** [comment]
That would be impossible...

**Comment by glowiak** [comment]
Why?

**Comment by Scrumplex** [comment]
I don't really see an easy way to achieve this. GitHub is not just for mods, it's for all kinds of projects. There is no easy way to tell if a repo is a Minecraft mod. Also it would be hard to figure out which Minecraft versions are supported and which loader is required.

**Comment by IceColdTray** [comment]
Idk how but this guy got it to work https://github.com/theRookieCoder/ferium

**Comment by flowln** [comment]
> Idk how but this guy got it to work https://github.com/theRookieCoder/ferium

From what I could gather, they use a Github rust API to go through a project's releases, looking for a release that contains a `.jar` file. If they find one, they assume it's a Minecraft mod and use it.

There's two main problems with doing something like this is PolyMC:
1. Is there a C++ API for Github that allows us to do such thing? If there is, would we be able to traverse the releases list without spamming that API?
2. To work on PolyMC, we would need to validate the JAR file as a Minecraft mod for a given version and mod loader. If we traverse the releases list searching for JAR files, assuming we could do so, we would need to download each and every JAR file we find, to validate it on the launcher (which would not require much work in the launcher itself tbh, since we already ready data from the JAR files themselves), but that's not really a nice thing to do right, both for the user's internet and Githubs' servers. Maybe it could be feasible if we only looked at the first JAR found, but I think that would be a pretty weird limitation for the user to have, and could generate a lot of confusion.

So, long story short, it could be doable, but it would probably either be severely limited in its scope / capabilities, or too much network resource-intensive for both sides of this feature. IMO, if you use mods from GitHub releases, your best bet is probably to try opening an issue on that repository asking for the author to place the mod on a mod distribution platform (like Modrinth or Curseforge).

**Comment by ZekeZDev** [comment]
Having a look at how ferium works this is 100% possible however there are limits to the api and I could see that becoming an issue unless we make users authenticate to github. There would also be no way for the user to search mods much like ferium they would need the repo owner and repo name. This could be useful once we implement auto update for mods meaning a user could have mods from github that auto update to latest releases.

**Comment by IceColdTray** [comment]
I would love to see this in the launcher but is this even really even possible?

**Comment by DioEgizio** [comment]
Btw wanted to say I found this lib: https://github.com/Kicer86/github_api so this is probably possible
Also packwiz is adding gh releases support, so we also have support for that in the .pw.toml spec
